astropy/wcs/src/unit_list_proxy.c

```diff
@@ -186,6 +187,59 @@ PyUnitListProxy_getitem(
   return result;
 }
 
+static PyObject *
+PyUnitListProxy_richcmp(
+    PyUnitListProxy *a,
+    PyUnitListProxy *b,
+    int op) {
+
+  int equal;
+  int status;
+  struct PyUnitListProxy *ax;
+  struct PyUnitListProxy *bx;
+   
+  
+  if ((op == Py_EQ || op == Py_NE) &&
+      PyObject_TypeCheck(b, &PyUnitListProxyType)) {
+      ax = &((PyUnitListProxy *)a)->pyobject;
+      bx = &((PyUnitListProxy *)b)->pyobject;
+          
+    
+    if (!PyObject_RichCompareBool(a->unit_class, b->unit_class,Py_EQ)||
+        a->size != b->size||
+        strncmp(a->array, b->array, 72) ) {
+        status = 0;
+        equal = 0;
+       }
+
+    else if (ax == NULL || bx == NULL){
+       status = 1;
```

Comment:
It seems like in case `status` is `1` you return `NotImplemented`, but this should probably be an error, right?

Reply:
Should I need to print a statement for this that argument is null or return `Py_RETUN_FALSE`

Reply:
Generally you set an exception and return `NULL` in case of failures. There is probably documentation about this, just to provide you some example from one of my own C extensions: https://github.com/MSeifert04/iteration_utilities/blob/v0.7.0/src/_itemidxkey.c#L47-L50

However I suspect `richcomparefunc` has some safe guards. You could check what CPython does in these kinds of functions (see for example https://github.com/python/cpython/tree/master/Objects)